astropy/modeling/parameters.py

```diff
@@ -746,3 +751,12 @@ def param_repr_oneline(param):
     if param.unit is not None:
         out = f"{out} {param.unit!s}"
     return out
+
+
+def _wrap_ufunc(ufunc):
+    def _wrapper(value, raw_unit=None, orig_unit=None):
+        if orig_unit is not None:
+            return ufunc(value) * orig_unit
+        return ufunc(value)
```

Comment:
Or we could choose to only support this behavior for dimensionless units? I proposed a suggestion under "Expected Behavior" in https://github.com/astropy/astropy/issues/14511.

Reply:
The purpose of the behavior in question really was to support storing parameter values under a different unit than what it is expressing it as. So really we should be solving the larger issue rather than adding yet another exception to the normal logic.